akka-docs/rst/scala/code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.persistence
+
+import java.nio.charset.Charset
+
+import akka.persistence.PersistentRepr
+import akka.persistence.PersistentRepr._
+import akka.persistence.journal.{EventSeq, EventAdapter}
+import com.typesafe.config._
+import scala.concurrent.duration._
+import org.scalatest.WordSpec
+import akka.actor.{Actor, ActorSystem}
+import akka.serialization.{SerializerWithStringManifest, Serializer, SerializationExtension}
+import akka.testkit.TestKit
+
+class PersistenceSchemaEvolutionDocSpec extends WordSpec {
+
+  val customSerializerConfig =
+    """
+      //#custom-serializer-config
+      akka.actor {
+        serializers {
+          my-payload = "docs.persistence.MyPayloadSerializer"
+          my-snapshot = "docs.persistence.MySnapshotSerializer"
+        }
+        serialization-bindings {
+          "docs.persistence.MyPayload" = my-payload
+          "docs.persistence.MySnapshot" = my-snapshot
+        }
+      }
+      //#custom-serializer-config
+    """
+
+  val system = ActorSystem("PersistenceSchemaEvolutionDocSpec", ConfigFactory.parseString(customSerializerConfig))
+  try {
+    SerializationExtension(system)
+  } finally {
+    TestKit.shutdownActorSystem(system, 10.seconds, false)
+  }
+
+}
+
+class ProtobufReadOptional {
+  object proto {
+    class SeatReserved {
+      def hasSeatType = false
+      def getCode = ""
+      def getSeatType = ""
+    }
+    object SeatReserved {
+      def newBuilder = new {
+        def setCode(any: Any): this.type = this
+        def setSeatType(any: Any): this.type = this
+        def build() = new {
+          def toByteArray: Array[Byte] = Array()
+        }
+      }
+      def parseFrom(any: Any) = new SeatReserved
+    }
+  }
+  
+  //#protobuf-read-optional-model
+  sealed abstract class SeatType { def code: String }
+  object SeatType {
+    def fromString(s: String) = s match {
+      case Window.code => Window
+      case Aisle.code => Aisle
+      case Other.code => Other
+      case _ => Unknown
+    }
+    case object Window  extends SeatType { override val code = "W" }
+    case object Aisle   extends SeatType { override val code = "A" }
+    case object Other   extends SeatType { override val code = "O" }
+    case object Unknown extends SeatType { override val code = ""  }
+
+  }
+  
+  case class SeatReserved(code: String, seatType: SeatType)
```

Comment:
what is the relation between this `code` and the `SeatType.code`? such reuse can be confusing